BookPlayer/Library/ItemList Screen/ItemListViewModel.swift

```diff
@@ -1311,6 +1311,43 @@ extension ItemListViewModel {
       )
     ))
   }
+
+  func getDownloadURL(for givenString : String) -> URL? {
```

Comment:
let's make this a throwable function, so we return a non-optional URL. I think we can cover the throwable errors with just a `invalidURL` one, which we already have the localized strings for. It would go something like:

```swift
guard 
  let givenURL = URL(string: givenString),
  let hostname = givenURL.host
else {
  throw SomeErrorEnum.invalidURL
}

switch hostname {
  case "drive.google.com":
    return try getGoogleDriveURL(for: givenURL)
  case "dropbox.com", "www.dropbox.com":
    return try getDropboxURL(for: givenURL)
  default:
    return givenURL
}
```

We could also fallback to the original URL for both cases if the try fails for either dropbox or google drive functions

Reply:
`getGoogleDriveURL()` does not have failure path -- if we are returning the original URL -- so didn't add a try there. But otherwise I believe I have incorporated your suggestion.